src/lucky/http2_compliance_handler.cr

```diff
@@ -0,0 +1,43 @@
+# HTTP/2 Compliance Handler
+#
+# This handler ensures all response headers comply with HTTP/2 naming conventions
+# by normalizing them to lowercase with dashes. It runs after all other processing
+# to ensure the final headers sent to the client are HTTP/2 compliant.
+#
+# Include this handler in your middleware stack to enable HTTP/2 header compliance:
+#
+# ```
+# middleware_stack = [
+#   # ... other middleware
+#   Lucky::HTTP2ComplianceHandler.new,
+# ]
+# ```
+class Lucky::HTTP2ComplianceHandler
```

Comment:
This entire class isn't necessary because Crystal already does this out of the box.

![image](https://github.com/user-attachments/assets/a13e0ce3-9e32-4ed8-96ec-763070ebbc51)

Reply:
Sweet, good to know. I asked it to add regression tests to ensure that incorrect headers weren't added in the future, so this is probably what triggered that addition.